### PR TITLE
Fix suggester

### PR DESCRIPTION
**Pull request by beowulfenator**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #216

### CODE REVIEW

**Comment by beowulfenator** [comment]
There are issues I'd like to discuss. This update is technically not backwards compatible, because the returned data is in a slightly different format. The `_suggest` endpoint used to return suggest data only, `_search` returns it all. So if in the past data could get accessed as

```php
$result['customer_name'][0]['options']
```
now it is
```php
$result['suggest']['customer_name'][0]['options']
```

We could return `$result['suggest']`, not `$result`, but that removes some useful data, for example, which shards/indices were successful, which failed.

The other problem is input data format. The `suggest()` method (and `query()` method, for that matter) are written to support both arrays and strings as input. If it's an array, it is `Json::encode()`'d. This creates some ugliness in the code, or otherwise I'd need to decode the input string.

Thoughts on this?

**Comment by samdark** [comment]
There are many installations running on 2.1 dev branch so these may break. While technically we haven't promised 2.1 release will be compatible with 2.1 dev API or 2.0, it's a good idea to keep the format unless there's a very good reason for a change.

**Comment by samdark** [comment]
How about keeping `suggest()` returning `$result['suggest']` but adding `search()` as well?

**Comment by beowulfenator** [comment]
I guess you are right, we could return `$result['suggest']`, and if full result is needed, then they could `search()`. What is your opinion on the other issue: building JSON body by hand?

**Comment by samdark** [comment]
I'm not sure what ugliness is that? The presence of encode itself?

**Comment by beowulfenator** [comment]
This:
```php
$body = '{"suggest":'.$suggester.',"size":0}';
```

**Comment by samdark** [comment]
Well, it's not that ugly by itself but the fact that there's no escaping or anything kind of is. I see no other way to achieve it though...

**Comment by beowulfenator** [comment]
Shall we merge it then?

File: tests/QueryTest.php [hunk 1]
```diff
@@ -436,4 +436,19 @@ public function testQueryWithWhere()
         $total = is_array($result['hits']['total']) ? $result['hits']['total']['value'] : $result['hits']['total'];
         $this->assertEquals(1, $total);
     }
+
+    public function testSuggest()
+    {
+        $cmd = $command = $this->getConnection()->createCommand();
```
Comment (Julian-B90) [inline tests/QueryTest.php:442]:
```suggestion
        $cmd = $this->getConnection()->createCommand();
```

Comment (beowulfenator) [reply tests/QueryTest.php]: Thank you

**Pull request merged**